src/main/java/org/opentripplanner/street/model/edge/StreetEdgeBuilder.java

```diff
@@ -0,0 +1,232 @@
+package org.opentripplanner.street.model.edge;
+
+import static org.opentripplanner.street.model.edge.StreetEdge.BACK_FLAG_INDEX;
+import static org.opentripplanner.street.model.edge.StreetEdge.BICYCLE_NOTHRUTRAFFIC;
+import static org.opentripplanner.street.model.edge.StreetEdge.CLASS_LINK;
+import static org.opentripplanner.street.model.edge.StreetEdge.HASBOGUSNAME_FLAG_INDEX;
+import static org.opentripplanner.street.model.edge.StreetEdge.MOTOR_VEHICLE_NOTHRUTRAFFIC;
+import static org.opentripplanner.street.model.edge.StreetEdge.ROUNDABOUT_FLAG_INDEX;
+import static org.opentripplanner.street.model.edge.StreetEdge.SLOPEOVERRIDE_FLAG_INDEX;
+import static org.opentripplanner.street.model.edge.StreetEdge.STAIRS_FLAG_INDEX;
+import static org.opentripplanner.street.model.edge.StreetEdge.WALK_NOTHRUTRAFFIC;
+import static org.opentripplanner.street.model.edge.StreetEdge.WHEELCHAIR_ACCESSIBLE_FLAG_INDEX;
+
+import org.locationtech.jts.geom.LineString;
+import org.opentripplanner.framework.i18n.I18NString;
+import org.opentripplanner.framework.i18n.NonLocalizedString;
+import org.opentripplanner.framework.lang.BitSetUtils;
+import org.opentripplanner.street.model.StreetTraversalPermission;
+import org.opentripplanner.street.model.vertex.StreetVertex;
+import org.opentripplanner.street.search.TraverseMode;
+
+public class StreetEdgeBuilder<B extends StreetEdgeBuilder<B>> {
+
+  // TODO(flamholz): do something smarter with the car speed here.
+  public static final float DEFAULT_CAR_SPEED = 11.2f;
+  public static final float DEFAULT_WALK_SAFETY_FACTOR = 1.0f;
+  private static final float DEFAULT_BICYCLE_SAFETY_FACTOR = 1.0f;
+
+  private StreetVertex from;
+  private StreetVertex to;
+  private LineString geometry;
+  private I18NString name;
+  private int millimeterLength;
+  private StreetTraversalPermission permission;
+  private boolean defaultLength;
+  private float carSpeed;
+  private float walkSafetyFactor;
+  private float bicycleSafetyFactor;
+  private short flags;
+
+  public StreetEdgeBuilder() {
```

Comment:
In the transit model we have static constructors like `StreetEdge.of()` which I quite like.

Reply:
- StreetEdge.of() would be convenient, but the static method would be inherited in the sub-classes (same problem as with the factory methods)
- The two vertices  are required already in the Edge superclass. In the transit model we tried not to duplicate non-null checks (see TripPattern/TripPatternBuilder)

Reply:
The static `of` init builder methods are convenient, but have the drawback that they introduce circular dependencies in inheritance hierarchies. In this case however; I think we can add it without introducing circular dependencies, since there is a 1-to-1 relationship between domain class and builder (`StreetEdge` <-> `StreetEdgeBuilder`, `AreaEdge` <-> `EreaEdgeBuilder` ...). For example, we can add `public static AreaEdgeBuilder of() {...}`  in `AreaEdge` (not in `StreetEdge`). 

The prefer place for integrity checks are in the constructor of the domain class, not the builder.